### PR TITLE
Add implementation of fs.ReadFile for io/fs.ReadFileFS interface

### PR DESCRIPTION
**Pull request by purpleidea**
- Adds the implementation of ReadFile in two different ways.
- Adds a test to make sure we fulfill the interface and that it
  works.

### CODE REVIEW

**Comment by purpleidea** [comment]
Neat project, thanks! I noticed it wasn't fulfilling this important interface, so I added it. If you like filesystem stuff and want to improve on my failings in https://github.com/purpleidea/mgmt/ which contains an Afero API I'd like to port away from, please let me know!

File: merged_fs_test.go [hunk 1]
```diff
@@ -103,6 +103,91 @@ func TestMergedFS(t *testing.T) {
...
+
+	// Merge all three zip files into a single FS, with zip1 being highest
+	// priority, and zip3 being lowest priority.
+	var m fs.FS = NewMergedFS(zip1, NewMergedFS(zip2, zip3))
+	merged, ok := m.(fs.ReadFileFS)
+	if !ok {
+		t.Errorf("Expected MergedFS to implement fs.ReadFileFS interface.")
+		return
+	}
+	expectedFiles := []string{
```
Comment (yalue) [inline merged_fs_test.go]: You can probably delete everything starting here (line 119) through line 159.  I understand that you want to call `fstest.TestFS` on the `ReadFileFS` instance, but fortunately the `fstest` package already includes special cases if the target FS implements the interface (for example, see [this line in the fstest code](https://cs.opensource.google/go/go/+/refs/tags/go1.22.0:src/testing/fstest/testfs.go;drc=6dc2d2aa6b9cbab9dc190276cb5a77b4b4a91cd3;l=531)).  Therefore, by implementing the ReadFileFS interface, the existing `TestMergedFS` function's call to `fstest.TestFS` will be sufficient for testing the new functionality.

(Though I still think it's good to have the explicit test that `test1.txt` produces the same contents when read through either method.)

File: merged_fs.go [hunk 2]
```diff
@@ -509,6 +509,59 @@ func (m *MergedFS) Open(path string) (fs.File, error) {
...
+// This is being added so that we fulfill the io/fs.ReadFileFS interface.
+// https://pkg.go.dev/io/fs#ReadFileFS
+func (m *MergedFS) ReadFile(name string) ([]byte, error) {
+	f, err := m.Open(name)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close() // ignore error
+
+	if true { // safer method in case the stdlib changes
```
Comment (yalue) [inline merged_fs.go]: Can we just keep the call to `io.ReadAll(f)` and remove the remaining dead code?

**Review by yalue — CHANGES_REQUESTED** [review]

**Comment by yalue** [comment]
Thanks for the change, with a new test and everything!  I just left a couple comments and I'll be happy to accept it after that.  It looks mgmt is quite an incredible project.  I'll take a look at it, and I do like filesystem stuff (at least in theory), though I doubt I'd be able to contribute much---my day job means I only have evenings and weekends free for maintaining my projects on github.

**Comment by purpleidea** [comment]
Sorry for the delay, I don't have lots of free time. Here's an updated version. Cheers.

**Comment by yalue** [comment]
Thanks, it's now in tagged version `v1.3.0`.

**Comment by purpleidea** [comment]
Thank you! HTH

(fwiw :scream: you can fast-forward these simple commits and especially for small projects it will keep your history cleaner)

**Pull request merged**